src/folding/nova/circuits.rs

```diff
@@ -330,55 +225,148 @@ where
             cmT.y,
         ]
         .concat();
-        CRHGadget::<C::ScalarField>::evaluate(crh_params, &input)
+        sponge.absorb(&input)?;
+        let bits = sponge.squeeze_bits(N_BITS_RO)?;
+        Ok(bits)
     }
 }
 
-impl<C: CurveGroup, FC: FCircuit<CF1<C>>> ConstraintSynthesizer<CF1<C>> for AugmentedFCircuit<C, FC>
+/// FCircuit defines the trait of the circuit of the F function, which is the one being executed
+/// inside the agmented F' function.
+pub trait FCircuit<F: PrimeField>: Clone + Copy + Debug {
+    /// returns a new FCircuit instance
+    fn new() -> Self;
+
+    /// computes the next state values in place, assigning z_{i+1} into z_i, and
+    /// computing the new z_i
+    fn step_native(
+        // this method uses self, so that each FCircuit implementation (and different frontends)
+        // can hold a state if needed to store data to compute the next state.
+        self,
+        z_i: Vec<F>,
+    ) -> Vec<F>;
+
+    /// generates the constraints for the step of F for the given z_i
+    fn generate_step_constraints(
+        // this method uses self, so that each FCircuit implementation (and different frontends)
+        // can hold a state if needed to store data to generate the constraints.
+        self,
+        cs: ConstraintSystemRef<F>,
+        z_i: Vec<FpVar<F>>,
+    ) -> Result<Vec<FpVar<F>>, SynthesisError>;
+}
+
+/// AugmentedFCircuit implements the F' circuit (augmented F) defined in
+/// [Nova](https://eprint.iacr.org/2021/370.pdf) together with the extra constraints defined in
+/// [CycleFold](https://eprint.iacr.org/2023/1192.pdf).
+#[derive(Debug, Clone)]
+pub struct AugmentedFCircuit<
+    C1: CurveGroup,
+    C2: CurveGroup,
+    GC2: CurveVar<C2, CF2<C2>>,
+    FC: FCircuit<CF1<C1>>,
+> where
```

Comment:
i) I think that for the moment we're fine with Nova+CycleFold, the logic gets simplier with CycleFold than with 'traditional' Nova, and we get the benefit of ending up with values on the Curve 1 instead of Curve 1 and Curve 2. If needed in the future we could extend it to have traditional-Nova.

ii) Regarding the `FCircuit`, this is the circuit that we're folding, the one that goes inside the `AugmentedFCircuit` ($F'$), which is what it is being folded in Nova+CycleFold (and same would apply for traditional-Nova). In other words, `FCircuit` trait is the trait that a user of this library would need to implement for their circuit to then start folding it using the Nova+CycleFold.

Reply:
Ohhh my bad! I understood the other way arround. Apologies!